apps/emqx_resource/test/emqx_connector_demo.erl

```diff
@@ -243,7 +243,7 @@ on_get_status(_InstId, #{pid := Pid}) ->
     end.
 
 spawn_counter_process(Name, Register) ->
-    Pid = spawn_link(?MODULE, counter_loop, []),
```

Comment:
In the happy case `on_stop` callback should take care of that.

https://github.com/emqx/emqx/blob/d44e683088b9264ed0531f446c398d57924034bb/apps/emqx_resource/test/emqx_connector_demo.erl#L77-L79

But in general, it might not be enough. Brought that back, instead tweaked the `on_stop/2` behaviour.